lib/phoenix/socket/transport.ex

```diff
@@ -379,6 +379,9 @@ defmodule Phoenix.Socket.Transport do
         :uri ->
           {:uri, fetch_uri(conn)}
 
+        :protocol ->
+          { :protocol, Plug.Conn.get_req_header( conn, "sec-websocket-protocol" ) }
```

Comment:
```suggestion
          {:protocol, Plug.Conn.get_req_header(conn, "sec-websocket-protocol")}
```

Reply:
My suggestion is to generalize this as `sec_headers`, so we can easily retrieve all security related headers instead of adding them one by one.

Reply:
The websocket javascript api *only* allows you set the 'sec-websocket-protocol' during the handshake. https://html.spec.whatwg.org/multipage/web-sockets.html#network so in this case it would only make sense to do the one. There are other header values defined in the RFC https://tools.ietf.org/html/rfc6455#section-4.3 but not in the spec. 

Unless you have something else in mind @josevalim